core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -2213,16 +2236,21 @@ object Parser {
       }
     }
 
+    /** capture parser p repeatedly, at least min times, at most max times
+      */
     final def repCapture[A, B](
         p: Parser[A],
         min: Int,
+        max: Int,
         state: State,
         append: Appender[A, B]
     ): Boolean = {
+      //require(max >= 0)
       var offset = state.offset
       var cnt = 0
-
-      while (true) {
+      //max == Int.MaxValue is a sentinel value meaning "forever"
+      val maxMinusOne = if (max == Int.MaxValue) max else max - 1
```

Comment:
Can we do this higher up? As it is, if you really mean Int.MaxValue you get infinity. If you do it higher up you would pass Int.MaxValue - 1 for that case and Int.MaxValue will be reserved for infinity.

That all said, maybe we should use Longs internally since 2 Billion really isn't totally enormous. Long.MaxValue is like infinity because no computer we are ever going to use will be able to run through that counter in any human scale time.

Reply:
It's moved up. Would you still like to switch to `Long`?

Reply:
I don't think we need that due to the point about String being limited to Int.MaxValue in length and this consuming at least one char per iteration.